client/src/net.rs

```diff
@@ -118,21 +113,31 @@ async fn handle_outgoing(
 /// Receive unordered messages from the server
 async fn handle_unordered(
     incoming: mpsc::UnboundedSender<Message>,
-    uni_streams: quinn::IncomingUniStreams,
+    connection: quinn::Connection,
 ) -> Result<()> {
-    let mut msgs = uni_streams
-        .map(|stream| async {
-            let stream = stream?;
-            codec::recv_whole::<proto::StateDelta>(2usize.pow(16), stream).await
-        })
-        .buffer_unordered(128);
-    // TODO: Don't silently die on parse errors
-    while let Some(msg) = msgs.try_next().await? {
-        // Ignore errors so we don't panic if the simulation thread goes away between checking
-        // `msgs` and here.
-        let _ = incoming.send(Message::StateDelta(msg));
+    loop {
+        let stream = connection.accept_uni().await;
+        let stream = match stream {
+            Err(quinn::ConnectionError::ApplicationClosed { .. }) => {
+                return Ok(());
+            }
+            Err(e) => {
+                return Err(e.into());
```

Comment:
It doesn't look like this return value goes anywhere. Might as well modify the function to not return anything, and consider sending an error on `incoming` here. Alternatively, handle errors at the callsite with the `tokio::spawn(async move { if let Err(e) = ... { ... });` pattern.